src/services/codefixes/fixMissingTypeAnnotationOnExports.ts

```diff
@@ -0,0 +1,1090 @@
+            {
+                pos: heritageExpression.getFullStart(),
+                end: realEnd,
+            },
+            heritageVariableName,
+            {
+                prefix: " ",
+            },
+        );
+        return [Diagnostics.Extract_base_class_to_variable];
+    }
+
+    interface ExpressionReverseChain {
+        element?: BindingElement;
+        parent?: ExpressionReverseChain;
+        expression: SubExpression;
+    }
+
+    const enum ExpressionType {
+        TEXT = 0,
+        COMPUTED = 1,
+        ARRAY_ACCESS = 2,
+        IDENTIFIER = 3,
+    }
+
+    type SubExpression =
+        | { kind: ExpressionType.TEXT; text: string; }
+        | { kind: ExpressionType.COMPUTED; computed: Expression; }
+        | { kind: ExpressionType.ARRAY_ACCESS; arrayIndex: number; }
+        | { kind: ExpressionType.IDENTIFIER; identifier: Identifier; };
+
+    function transformDestructuringPatterns(bindingPattern: BindingPattern): DiagnosticOrDiagnosticAndArguments | undefined {
+        const enclosingVariableDeclaration = bindingPattern.parent as VariableDeclaration;
+        const enclosingVarStmt = bindingPattern.parent.parent.parent as VariableStatement;
+        if (!enclosingVariableDeclaration.initializer) return undefined;
+
+        let baseExpr: ExpressionReverseChain;
+        const newNodes: Node[] = [];
+        if (!isIdentifier(enclosingVariableDeclaration.initializer)) {
+            // For complex expressions we want to create a temporary variable
+            const tempHolderForReturn = factory.createUniqueName("dest", GeneratedIdentifierFlags.Optimistic);
+            baseExpr = { expression: { kind: ExpressionType.IDENTIFIER, identifier: tempHolderForReturn } };
+            newNodes.push(factory.createVariableStatement(
+                /*modifiers*/ undefined,
+                factory.createVariableDeclarationList(
+                    [factory.createVariableDeclaration(
+                        tempHolderForReturn,
+                        /*exclamationToken*/ undefined,
+                        /*type*/ undefined,
+                        enclosingVariableDeclaration.initializer,
+                    )],
+                    NodeFlags.Const,
+                ),
+            ));
+        }
+        else {
+            // If we are destructuring an identifier, just use that. No need for temp var.
+            baseExpr = { expression: { kind: ExpressionType.IDENTIFIER, identifier: enclosingVariableDeclaration.initializer } };
+        }
+
+        const bindingElements: ExpressionReverseChain[] = [];
+        if (isArrayBindingPattern(bindingPattern)) {
+            addArrayBindingPatterns(bindingPattern, bindingElements, baseExpr);
+        }
+        else {
+            addObjectBindingPatterns(bindingPattern, bindingElements, baseExpr);
+        }
+
+        const expressionToVar = new Map<Expression, Identifier>();
+
+        let i = 0;
+        while (i < bindingElements.length) {

```

Comment:
Nit, but why not a regular for loop here instead of a while?

Reply:
Converted to a for loop.

Reply:
Where? In a PR on the fork? (I didn't see a push, just checking in case it's a forgotten push)

Reply:
Ah, yes, I see the PR.